addon/-private/class/modifier-manager.ts

```diff
@@ -0,0 +1,70 @@
+import Ember from "ember";
+import { capabilities } from "@ember/modifier";
+import { set } from "@ember/object";
+import { schedule } from "@ember/runloop";
+
+import ClassBasedModifier, {
+  DESTROYING,
+  DESTROYED,
+  InTeardown,
+} from "./modifier";
+import { ModifierArgs } from "ember-modifier/-private/interfaces";
+
+class ClassBasedModifierManager {
+  capabilities = capabilities("3.13");
+
+  createModifier(
+    factory: { owner: unknown; class: typeof ClassBasedModifier },
+    args: ModifierArgs
+  ) {
+    let { owner, class: modifier } = factory;
+
+    return new modifier(owner, args);
+  }
+
+  installModifier(instance: ClassBasedModifier, element: Element) {
+    instance.element = element;
+    instance.didReceiveArguments();
+    instance.didInstall();
+  }
+
+  updateModifier(instance: ClassBasedModifier, args: ModifierArgs) {
+    // TODO: this should be an args proxy
+    set(instance, "args", args);
+    instance.didUpdateArguments();
+    instance.didReceiveArguments();
+  }
+
+  // Takes `Teardown<ClassBasedModifier>` to correctly model the type of
+  // `element` at this point in the lifecycle. This is safe because this manager
+  // is what *defines* that lifecycle behavior.
+  destroyModifier(instance: InTeardown<ClassBasedModifier>) {
+    instance.willRemove();
+    instance.element = null;
+
+    if (instance[DESTROYING]) {
+      return;
+    }
+
+    let meta = Ember.meta(instance);
+
+    meta.setSourceDestroying();
+    instance[DESTROYING] = true;
+
+    schedule("actions", instance, instance.willDestroy);
+    schedule("destroy", undefined, scheduleDestroy, instance, meta);
```

Comment:
Can you create an issue to remove all of this code in favor of using the destroyables RFC (emberjs/rfcs#580) by way of https://github.com/ember-polyfills/ember-destroyable-polyfill?

Reply:
Definitely!

Reply:
Tracking conversion to using the polyfill in #24.